### ISSUE TITLE
About those donations

### ISSUE THREAD

**Issue by braver** [SublimeLinter/SublimeLinter#820]
So, we're asking for donations at the bottom of the README and apparently it collect a few dollars sometimes. Not much, but also not nothing. I haven't spoken groteworld in a while (he maintains the accounts), so I'm not sure what came in in the last few months. However, I also wouldn't know what to do with it. We've gotten outsize contributions from @pykong and @kaste over the past few months, but how do I distribute that? Others made contributions too that may have been significant parts of their spare time (@FichteFoll @mandx @gerardroche), although perhaps not as much in absolute terms. Also, how do you distribute donations that come in in the future? Is a huge contribution today (or 2 years ago for that matter) not worth a share of the donations 3 months from now?

There are some costs for the domain name (sublimelinter.com, which is a wast of money IMO but whatever), so some of the donations could go towards that. But I really have no idea what to do with the rest in a way that's fair. This isn't someone's pet project, it's a community effort.

We may be talking about just $2,- for all I know, but if we're asking for donations we should also know what to do with it and I certainly don't need or event want any of it. 

So this is a request for opinion. Anyone have anything smart to say about this, I'm all ears.

My personal idea would be to find a good charity and send it there. It should be something global and something to do with education. I have a soft spot for the Jane Goodall Foundation, which doesn't just do wild life preservation but a lot of education as well.

**Comment by pykong** [comment]
I feel contributions should one and foremost cover the expenses of the project itself.
Expenses may come in different forms be it hosting cost, or eventual bug bounties.

I actually do not like the idea of us serving as a donation proxy for a charity.
At least keep it relevant to FOSS, because that is what we are doing here and also may better connect with the average person using SublimeLinter, ie the potential average donor, who eventually is a computer guy.

What about example the [Python Software Foundation](https://www.python.org/psf/)?
After all we wrote a lot of python for SL4, didn't we?

**TL:DR**:
- Rather use donation for own expenses.
- Do not use donations for cheering up contributors.
- If donations are directed towards charity, keep it relevant to the work we did.
- In the latter case the **Python Software Foundation** is a candidate.

**Comment by mandx** [comment]
The problem with the domain name, is that it already exists, people may be referencing it, others may have it bookmarked, etc. And I would assume it contributes a lot with page ranking when searching online. Since STLinter is a "framework" to integrate linters with Sublime, documentation (and access to it) are really important.

Regarding my participation: the fact the STLinter exists and it is actively developed/maintained, that's the best retribution I could ever have :)

In general, I agree with @pykong's TL:DR.

**Comment by FichteFoll** [comment]
I also agree with this. 

Previously, this has been a single effort by aparajita, who has asked for donations in order to keep developing this large package ecosystem, which is totally fair. But since he stepped down and groteworld hasn't done much in recent time either, this has become a large community effort, which I personally find absolutely great.
@pykong's suggestions for spending donated money primarily on project expenses (probably only the domain) and bounties sounds good to me. Bounties haven't been tested for this project, so we don't know whether this use case will come up in the future, but I guess if there is a lot of money in the bank, finding a way to spend that could be useful. I also do not like the idea of acting as a donation proxy as people probably want their support to be more direct, but I believe that supporting technologies we use, such as the PSF, isn't so bad if there really isn't anything else to put money towards.

By the way, I think the only reason why I noticed this and started contributing to discussions, code and code reviews was because of a discussion about it on Discord, so I think that's great.

**Comment by braver** [comment]
> The problem with the domain name, is that it already exists, people may be referencing it, others may have it bookmarked, etc. And I would assume it contributes a lot with page ranking when searching online.

That's a good point. I would kinda expect packagecontrol.io to be the entry point for most people looking for information about a package though. But the domain name is fine, we'll probably keep it.

Thanks for thoughts and suggestions so far already. In any case it's good to see we seem to have a similar position on this. 

I definitely won't be keeping the donation link around as a proxy for charity, that indeed doesn't sound like a good idea. It's more about probably/maybe already having a few dollars sitting in the account and what to do with that. The PSF in particular is sponsored to a level where individual (and probably small) donations don't make sense. There might be a different non-profit out there that would benefit from a donation, so that might be option.

I don't like bounties because the money needlessly complicates things. What if 2 people open up a PR with similar proposals? This is open source, if no-one cares enough to fix something for free than it's not important enough to get fixed. If it's so complicated the active contributors can't figure it out, they probably also can't maintain the solution. It's a good fit for some things, I don't think it's a good fit for an editor plugin.

Again, if anyone has another good idea I'm all ears. Just letting the money sit there for expenses is also not a bad idea, unless it's something like $100 (which I would find amazing, but also unlikely).

**Comment by groteworld** [comment]
Sorry, I've been MIA again. The coffer is actually $126 as of today. Since I've mostly stepped down in favor of @braver's control of the project, I believe they should inherit control of this as well so it's not under my usage/absence/negligence. While I don't have a strong say for what this money should go to, I've tried to give donations out to large contributions in the past. As for tech costs: the domain has always been considered a yearly donation from @jswartwood. I'm unsure of his wishes, if he sees it as a donation, or if he would like to resell it to us, but something to keep in mind.

**Comment by braver** [comment]
Oh shit, that's actually a serious amount of money.

**Comment by gerardroche** [comment]
I don't consider myself part of of share, even if it was lot of money. It's nonetheless an interesting problem: how to divide donations to contributors? I'd imagine it can cause some issues when really big amounts of money are involved.

It's a pity about the domain, because it probably isn't needed. If there are hosting costs, I'd recommend moving the site to github pages. I don't think there's much of a downside. You could setup `sublimelinter.github.com` and start pointing everything there, you still won't be able to let go of the domain, but you might further on down the road.

**Comment by groteworld** [comment]
Hosting is just readthedocs so those are also free.

**Comment by braver** [comment]
@kaste I would love to get your opinion here as well, even if it’s just 🤷🏼‍♂️

Considering there is actually a sizable amount the coffer I think the correct way to go about it now is to split it amongst the recent big contributors, excluding myself. This is the best match with the existing communication about it. We’ll be dropping SL4 very soon now, some time after that event and after @groteworld transfers the account I’ll be in touch with all involved about the details.

I haven’t decided on where to go with this in the future.

**Comment by kaste** [comment]
The people donating had probably the intention that 'we' actually get the money. Giving it to greenpeace is a jerks move. It is the equivalent of not taking a thank you, a gift/present. Sometimes you get a coffee or beer for free in real life as well.

**Comment by braver** [comment]
Excellent, definitely don't disagree with that. Ok, we're done here, thanks all.
